main.go

```diff
@@ -73,11 +76,11 @@ func main() {
 
 }
 
-func gpioPing(pin rpio.Pin) ping {
+func gpioPing(pin gpio.PinIO) ping {
```

Comment:
Not needed here , but this is a perfect example of where you'd define your own Pin Interface and use it

```go
// define the interface **you** want
type GodorPin Interface {
  isLow() bool
}

// adapt the struct you want to meet your interface
func (p gpio.PinIO) isLow() bool {
  return p.Low
}

// your code now documents your interface, and uses it
func gpioPing (pin GodorPin) ping {
  isOpen := !pin.isLow()
}
```